File: java-design-pattern/src/main/java/intro/to/design/patterns/using/javalang/factorymethod/FactoryMethodMain.java
package intro.to.design.patterns.using.javalang.factorymethod;

import intro.to.design.patterns.using.javalang.factorymethod.creditcard.CreditCardFactory;
import intro.to.design.patterns.using.javalang.factorymethod.framework.Factory;
import intro.to.design.patterns.using.javalang.factorymethod.framework.Product;
import intro.to.design.patterns.using.javalang.factorymethod.idcard.IDCardFactory;

import java.util.ArrayList;
import java.util.Arrays;

public class FactoryMethodMain {
    public static void main(String[] args) {
        Factory idCardFactory = new IDCardFactory();
        Factory creditCardFactory = new CreditCardFactory();
        ArrayList<Product> cards = new ArrayList<>(Arrays.asList(
                idCardFactory.create("Youngjin Kim"),
                idCardFactory.create("Heunmin Son"),
                creditCardFactory.create("Kane Watson"),
                creditCardFactory.create("Draco Malfoy"),
                creditCardFactory.create("John Doe")
        ));
        System.out.println();

        cards.forEach(Product::use);
    }
}

File: typescript-design-pattern/src/into-to-design-patterns-using-java-lang/factory-method/factory-method-main.ts
import Output from '../../utils/stdin-stdout/output'
import CreditCardFactory from './credit-card/credit-card-factory'
import Factory from './framework/factory'
import { Product } from './framework/product'
import IdCardFactory from './id-card/id-card-factory'

const factoryMethodMain = () => {
  const idCardFactory: Factory = new IdCardFactory()
  const creditCardFactory: Factory = new CreditCardFactory()
  const cards: Product[] = []
  cards.push(idCardFactory.create('Youngjin Kim'))
  cards.push(idCardFactory.create('Heunmin Son'))
  cards.push(creditCardFactory.create('Kane Watson'))
  cards.push(creditCardFactory.create('Draco Malfoy'))
  cards.push(creditCardFactory.create('John Doe'))
  Output.log()

  cards.forEach((card) => card.use())
}

export default factoryMethodMain
